netplugin/netd.go

```diff
@@ -242,6 +242,33 @@ func processEpState(netPlugin *plugin.NetPlugin, opts cliOpts, epID string) erro
 	return err
 }
 
+//processBgpEvent processes Bgp neighbor add/delete events
+func processBgpEvent(netPlugin *plugin.NetPlugin, opts cliOpts, hostID string,
+	isDelete bool) (err error) {
+
+	if opts.hostLabel != hostID {
+		log.Errorf("Ignoring Bgp Event on this host")
+		return
+	}
+	netPlugin.Lock()
+	defer func() { netPlugin.Unlock() }()
+
+	operStr := ""
+	if isDelete {
+		err = netPlugin.DeleteBgp(hostID)
+		operStr = "delete"
+	} else {
+		err = netPlugin.AddBgp(hostID)
+		operStr = "create"
+	}
+	if err != nil {
+		log.Errorf("Bgp operation %s failed. Error: %s", operStr, err)
+	} else {
+		log.Infof("Bgp operation %s succeeded", operStr)
+	}
+	return
```

Comment:
I agree.

On 11 May 2016, at 7:21, jojimt wrote:

>> +	defer func() { netPlugin.Unlock() }()
>> +
>> +	operStr := ""
>> +	if isDelete {
>> +		err = netPlugin.DeleteBgp(hostID)
>> +		operStr = "delete"
>> +	} else {
>> +		err = netPlugin.AddBgp(hostID)
>> +		operStr = "create"
>> +	}
>> +	if err != nil {
>> +		log.Errorf("Bgp operation %s failed. Error: %s", operStr, err)
>> +	} else {
>> +		log.Infof("Bgp operation %s succeeded", operStr)
>> +	}
>> +	return
>
> Interesting. One of the golang "features" I would rather stay away 
> from...
>
> ---
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/contiv/netplugin/pull/357/files/80c6ae0e2e635a996599f154e91f7e63d715abb4#r62856652